### ISSUE TITLE
DataBase not working on mobile device

### ISSUE THREAD

**Issue by manojy5951** [dfahlander/Dexie.js#192]
When using dexiejs for Local Storage in my mobile app the app works fine in local browser(chrome) but when i load the app in mobile device it show no data 
I am using predefined data that is first stored and then retrieved 
And I am using Dexies in my angular controllers

**Comment by dfahlander** [comment]
Need more info: platform and OS. Maybe related to #170.

**Comment by manojy5951** [comment]
I am developing for android  on windows 10

**Comment by dfahlander** [comment]
Are you using visual studio community, vscode or other IDE? If vs community, Is your project an Apache cordova project?

**Comment by manojy5951** [comment]
yes sir i am using vs2015  and the project is apache cordova project

**Comment by dfahlander** [comment]
Ok, thanks! Now it's easier for anyone to help you when we know the exact stack you're on. Unfortunately, I haven't exprerience with using the indexedDB API on apache cordova. Anyone else who knows?

**Comment by muthuveerappan** [comment]
Check the same in browser (windows 10 - browser edge/ie) and then later you can check on phone too ... May be few browsers need shims/polyfills

**Comment by muthuveerappan** [comment]
apache cordova on android should use chrome/chromium ., where as in windows it will use a different webview / html rendering engine / browser.

**Comment by Critical-Impact** [comment]
If you are targeting iOS make sure you include the indexedDB shim before the dexie JS files. Also I found that I had to have a line like 
var database = new Dexie("secondaryDatabase", {indexedDB: window.shimIndexedDB})

On Android, I find it just works without the shim so I'd suggest using platform detection to have a conditional for specifying when to use the shim.

Also, I cant say I have any experience with cordova specific plugins that might override websql or indexed DB so I cant comment on them but using the webview's indexedDB(android) and websql(ios) seems to work fine.
